File: app/src/main/java/com/sonu/app/splash/data/network/NetworkDataManager.java
package com.sonu.app.splash.data.network;

import com.sonu.app.splash.model.unsplash.Photo;
import com.sonu.app.splash.model.unsplash.PhotoStats;
import com.sonu.app.splash.model.unsplash.User;

import io.reactivex.Observable;

/**
 * Created by amanshuraikwar on 18/12/17.
 */

public interface NetworkDataManager {

    Observable<Photo> getPhotoDescription(String photoId);

    Observable<User> getUserDescription(String username);

    Observable<PhotoStats> getPhotoStats(String photoId);
}

File: app/src/main/java/io/github/amanshuraikwar/splash/data/network/NetworkDataManager.kt
package io.github.amanshuraikwar.splash.data.network

import io.github.amanshuraikwar.splash.model.Photo

/**
 * Data Manager for the content fetched from the network.
 * This is the single entry point to fetch data from the network.
 *
 * @author Amanshu Raikwar
 * Created by amanshuraikwar on 30/04/18.
 */
interface NetworkDataManager {

    /**
     *
     */
    fun getAllPhotos(page: Int, orderBy: String, perPage: Int): List<Photo>?

    /**
     *
     */
    fun getCurartedPhotos(page: Int, orderBy: String, perPage: Int): List<Photo>?
}